stdlib/contrib/qxip/logql/logql.flux

```diff
@@ -0,0 +1,87 @@
+// Package logql provides functions meant to simplify common logql queries.
+//
+// The primary function in this package is `logql.query_range()`
+//
+// ## Metadata
+// introduced: NEXT
+//
+package logql
+
+
+import "csv"
+import "date"
+import "experimental"
+import "experimental/http/requests"
+
+// query_range queries data from a specified LogQL query within given time bounds,
+// filters data by query, timerange, and optional limit expressions.
+//
+// ## Parameters
+// - url: LogQL/qryn URL.
+// - path: LogQL query_range API path.
+// - limit: Query limit. Default is 100.
+// - query: LogQL query to execute.
+// - start: Earliest time to include in results. Default is `-1h`.
+//
+//   Results include points that match the specified start time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `-1h`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
+// - end: Latest time to include in results. Default is `now()`.
+//
+//   Results exclude points that match the specified stop time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `now()`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
+// - step: Query stepping. Default is 10.
+// - orgid: Optional Organization Id for partitioning.
+//
+// ## Examples
+// ### Query specific fields in a measurement from LogQL/qryn
+// ```no_run
+// import "contrib/qxip/logql"
+//
+// logql.query_range(
+//     url: "http://qryn:3100",
+//     path: "/loki/api/v1/query_range",
+//     start: -1h,
+//     end: now(),
+//     query: "{job=\"dummy-server\"}",
+//     limit: 100,
+// )
+// ```
+//
+// ## Metadata
+// tags: inputs
+//
+query_range = (
+        url="http://127.0.0.1:3100",
+        path="/loki/api/v1/query_range",
+        query="",
+        limit=100,
+        step=10,
+        start=-1h,
+        end=now(),
+        orgid="",
+    ) =>
+    {
+        dstart = date.sub(from: start, d: 0d)
+        dend = date.sub(from: end, d: 0d)
```

Comment:
The way this is written, the `start` and `end` won't really do anything. This is currently subtracting a `0d` duration from each of the provided timestamps, so they return the timestamps unchanged. Also, this makes it so the `start` and `stop` parameters can only be times. Durations won't work this way.

To get the functionality you're looking for, I think instead of using `date.sub` here, you should use [`date.time`](https://docs.influxdata.com/flux/v0.x/stdlib/date/time/). This will return a time value when provided either a date or a duration. This will _not_ support unix timestamps (ints or uints).

```suggestion
        dstart = date.time(t: start)
        dend = date.time(t: end)
```

Reply:
This was/is intended as a formatting workaround for the date fields to work through the input. I'm actually only using the conversion to get the desired format output in ` [string(v: uint(v: dstart))]`, hence the 0d subtraction. Using RFC strings is not supported uniformly yet, so the microsecond format is the current requirement. If there's a better method to avoid the double-conversion, I'd love to learn how!